src/vulnpy/trigger/sqli.py

```diff
@@ -1,28 +1,29 @@
 import sqlite3
 
-db_connection = sqlite3.connect(":memory:")
```

Comment:
I'm not totally sure about this, but I think there's a chance that the `:memory:` database resets every time you make a new connection to it. That might be related to the test failures

Reply:
according to https://www.sqlite.org/inmemorydb.html:
> The database ceases to exist as soon as the database connection is closed. Every :memory: database is distinct from every other. So, opening two database connections each with the filename ":memory:" will create two independent in-memory databases.

So the solution you need here might be a little more complicated than opening several new connections to in-memory databases. It's possible that we need to just make an actual file-based database, but I'd prefer to avoid this if there's a way around it